src/Keboola/DbExtractor/Configuration/NodeDefinition/FirebirdTablesNode.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor\Configuration\NodeDefinition;
+
+use Keboola\DbExtractorConfig\Configuration\NodeDefinition\TablesNode;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+
+class FirebirdTablesNode extends TablesNode
+{
+    protected function init(): void
+    {
+        // @formatter:off
+        $this
+            ->prototype('array')
+            ->validate()->always(function ($v) {
+                if (isset($v['query']) && $v['query'] !== '' && isset($v['table'])) {
+                    throw new InvalidConfigurationException('Both table and query cannot be set together.');
+                }
+                if (isset($v['query']) && $v['query'] !== '' && isset($v['incrementalFetchingColumn'])) {
+                    $message = 'Incremental fetching is not supported for advanced queries.';
+                    throw new InvalidConfigurationException($message);
+                }
```

Comment:
tohle tady neni potreba kdyz `incrementalFetching` neni implementovane.  ale stejne nic to nerozbije

Reply:
máš pravdu - nebyla tam potřeba ani validace na `table` tak jsem to celé `validation` smazal a query dal `isRequired`